README.md

```diff
@@ -55,22 +55,23 @@ LGPL code.
 
 ## FEATURES
 
-RIOT is based on a microkernel architecture, and provides features including,
-but not limited to:
```

Comment:
This may be a bit controversial. But I 100% agree that RIOT-OS is not and never has been using a micro kernel architecture.

I recall that Andrew Tanenbaum gave a relatively lax interpretation of what a microkernel architecture is. The absence of any MMU / MPU like feature, according to him, doesn't rule out that an OS does check the box of a microkernel architecture. If I recall correctly, he said that MINIX was developed initially on pre-MMU systems with the anticipation of an MMU, so that making use of the MMU to actually provide the benefits of a microkernel architecture was relatively easy.

However, we with RIOT are now in a situation that MPUs are there. Still, the effort to make use of it in a microkernel point of view (or even in a userspace vs kernelspace point of view) is still significant. To me, this is a good indicator that RIOT is not using a microkernel architecture, even with the relatively lax point of view that actual use of MMU (or MPU) is not needed for the claim.

Reply:
I guess we all agree this qualification is not 100% fitting, but after all these years, and after pitching it here & there (including to top experts such as Tanenbaum as recalled by @maribu) it seems it remains the best "single term" description for the arch. For a longer explanation (which does not really belong right here) we could maybe open a separate wikipage / doc entry on this, what do you think?

Reply:
What makes you think the arch is a microkernel? Having a hand-wavy explanation to claim something that is not true does not build trust. How would you react to a car salesman selling you an electric¹ car?

[1] lead acid battery used to start the internal combustion engine